exit_gtk.py

```diff
@@ -0,0 +1,222 @@
+
+	def _key_press_event(self, widget, event):
+		keyval = event.keyval
+		keyval_name = gtk.gdk.keyval_name(keyval)
+		state = event.state
+		alt = (state & gtk.gdk.MOD1_MASK)
+		# Cancel shortcut
+		if alt and keyval_name == 'l':
+			self.send_to_dbus('Logout')
+		# Suspend shortcut
+		elif alt and keyval_name == 's':
+			self.send_to_dbus('Suspend')
+		# Reboot Shortcut
+		elif alt and keyval_name == 'b':
+			self.send_to_dbus('Reboot')
+		# Poweroff shortcut
+		elif alt and keyval_name == 'p':
+			self.send_to_dbus('PowerOff')
+		# Hibernate Shortcut
+		elif alt and keyval_name == 'i':
+			self.send_to_dbus('Hibernate')
+		# Hybrid Sleep Shortcut
+		elif alt and keyval_name == 'y':
+			self.send_to_dbus('HybridSleep')
+		# Cancel Shortcut
+		elif keyval_name == 'Escape':
+			self.destroy()
+		elif keyval_name == 'Alt':
+			self.show_button_labels()
+		else:
+			return False
+		return True
+		
+	def destroy(self, widget=None, event=None, data=None):
+		self.window.hide_all()
+		gtk.main_quit()
+
+	def __init__(self, button_values, exit_bus, theme, theme_entries, style_path):
+		# Attributes
+		self.theme_entries = theme_entries
+		self.window = gtk.Window(gtk.WINDOW_TOPLEVEL)
+		try:
+			self.dialog_height = int(self.theme_entries['dialog_height'])
+		except:
+			exit_log.warn("dialog_height not set. Settins a default of 64 pixels.")
+			self.dialog_height = 64
+		# Get the screen width under the cursor
+		screen_width = 800 # fallback width
+		try:
+			display=gtk.gdk.Display(gtk.gdk.get_display())
+			screen, x, y, flags=display.get_pointer()
+			curmon = screen.get_monitor_at_point(x, y)
+			_, _, screen_width, _ = screen.get_monitor_geometry(curmon)
+		except:
+			exit_log.info('Error constructing UI. Not running under X')
+		finally:
+			del x, y, display, screen, curmon
+		exit_log.debug("Detected screen_width is " + str(screen_width))
+		try:
+			self.width_adjustment = float(self.theme_entries['window_width_adjustment'])
+		except:
+			exit_log.warn("window_width_adjustment not a float. Please check config. Defaulting to 0.5")
+		if self.width_adjustment > 0:
+			self.dialog_width = int( screen_width * self.width_adjustment)
+		if self.dialog_width > screen_width:
+			self.dialog_width = screen_width
+		exit_log.debug("Dialog width is set to " + str(self.dialog_width))
+		# Format the window.
+		self.window.set_name('Bunsen Exit')
+		self.window.set_decorated(False)
+		self.window.connect("delete_event", self.destroy)
+		self.window.connect("destroy_event", self.destroy)
+		self.window.connect("key-press-event", self._key_press_event)
+		self.window.set_resizable(False)
+		self.window.set_keep_above(True)
+		self.window.stick()
+		self.window.set_position(gtk.WIN_POS_CENTER)
+		windowicon = self.window.render_icon(gtk.STOCK_QUIT, gtk.ICON_SIZE_DIALOG)
+		self.window.set_icon(windowicon)
+		self.window.modify_bg(gtk.STATE_NORMAL, gtk.gdk.color_parse(self.theme_entries['window_background_normal']))
```

Comment:
Same. On my Arch system (bunsen-exit never ran here), this leads to a gtk segfault because `self.theme_entries` does not (yet) have a key `window_backgrund_normal`. Trace:
```
__init__.py(1341):         if (found == 0) and raiseExceptions and not self.manager.emittedNoHandlerWarning:
exit_gtk.py(88):                self.window.set_name('Bunsen Exit')
exit_gtk.py(89):                self.window.set_decorated(False)
exit_gtk.py(90):                self.window.connect("delete_event", self.destroy)
exit_gtk.py(91):                self.window.connect("destroy_event", self.destroy)
exit_gtk.py(92):                self.window.connect("key-press-event", self._key_press_event)
exit_gtk.py(93):                self.window.set_resizable(False)
exit_gtk.py(94):                self.window.set_keep_above(True)
exit_gtk.py(95):                self.window.stick()
exit_gtk.py(96):                self.window.set_position(gtk.WIN_POS_CENTER)
exit_gtk.py(97):                windowicon = self.window.render_icon(gtk.STOCK_QUIT, gtk.ICON_SIZE_DIALOG)
exit_gtk.py(98):                self.window.set_icon(windowicon)
exit_gtk.py(99):                self.window.modify_bg(gtk.STATE_NORMAL, gtk.gdk.color_parse(self.theme_entries['window_background_normal']))
 --- modulename: trace, funcname: _unsettrace
trace.py(80):         sys.settrace(None)
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/usr/lib/python2.7/trace.py", line 819, in <module>
    main()
  File "/usr/lib/python2.7/trace.py", line 807, in main
    t.runctx(code, globs, globs)
  File "/usr/lib/python2.7/trace.py", line 513, in runctx
    exec cmd in globals, locals
  File "bunsen-exit", line 160, in <module>
    sys.exit(main())
  File "bunsen-exit", line 154, in main
    bl_exit_gtk = exit_gtk.ExitGtk(button_values, exit_bus, theme, theme_entries, style_path)
  File "exit_gtk.py", line 99, in __init__
    self.window.modify_bg(gtk.STATE_NORMAL, gtk.gdk.color_parse(self.theme_entries['window_background_normal']))
```

Reply:
FYI: How I obtained those traces:
```
python2.7 -m trace --trace ./bunsen-exit
```